### DISCUSSION TITLE
Real-time video encoding of the simulation

### DISCUSSION THREAD

**Discussion by sasmaster** [microsoft/AirSim#3492] [Ideas]
Hi All, I am new here. I would like to know if there is a plan to support viewport video recording to file. I see the AirSim currently allows only grabbing WebRTC stream. If there is no such a plan then maybe I will extend the built int NvEncoder to write into file instead. Is this feature important to anyone else?

**Comment by rajat2004** [comment]
It'll be a useful addition for sure, but from what I understand Pixel Streaming isn't a part of AirSim itself, and is a completely separate feature of Unreal Engine. So do you mean modifying the Unreal Engine or Pixel Streaming code to record to file?
Other option would be to maybe use some Python libraries or tools like FFmpeg or GStreamer to capture the stream and record. Note that I haven't myself tried it.

**Reply by sasmaster** [reply]
I actually checked the encoder used by the pixel streamer. It's not good for files , also the NVENC version is pretty old. I wrote a plugin with separate encoder. I need to clean it up a bit before sharing here.
